### ISSUE TITLE
Set up configuration management system

### ISSUE THREAD

**Issue by RobotPsychologist** [Blood-Glucose-Control/nocturnal-hypo-gly-prob-forecast#267]
#### Hierarchical Configuration Structure
- **Model Configs**: Model-specific parameters, architecture settings
- **Training Configs**: Distributed training settings, optimization parameters
- **Data Configs**: Dataset specifications, preprocessing parameters
- **Experiment Configs**: End-to-end experiment definitions

#### Configuration Composition
```yaml
# configs/experiments/nocturnal/ttm_point_forecast.yaml
model:
  type: ttm
  config: !include ../models/ttm/fine_tune.yaml
  
training:
  config: !include ../training/multi_gpu_ddp.yaml
  
data:
  datasets: !include ../data/multi_dataset.yaml
  
evaluation:
  config: !include ../evaluation/standard_metrics.yaml

**Comment by RobotPsychologist** [comment]
## Configuration Examples

### Model Configuration
```yaml
# configs/models/ttm/fine_tune.yaml
model_type: ttm
architecture:
  d_model: 512
  n_heads: 8
  n_layers: 6
  context_length: 512
  prediction_length: 96

training:
  strategy: fine_tuning
  base_model_path: "huggingface/ttm-1.5B"

lora:
  enabled: true
  rank: 16
  alpha: 32
  target_modules: ["q_proj", "v_proj"]

optimization:
  optimizer: adamw
  learning_rate: 1e-4
  weight_decay: 0.01
  warmup_steps: 1000
```

### Experiment Configuration
```yaml
# configs/experiments/nocturnal/ttm_vs_chronos.yaml
experiment_name: "TTM vs Chronos Nocturnal Forecasting"
experiment_type: nocturnal

models:
  - type: ttm
    config: !include ../../models/ttm/fine_tune.yaml
  - type: chronos
    config: !include ../../models/chronos/base.yaml

data:
  datasets:
    - kaggle_bris_t1d
    - aleppo
  splits:
    temporal_holdout: 0.2  # Last 20% of data
    patient_holdout: 0.15  # 15% of patients

training:
  distributed: true
  gpu_strategy: ddp
  max_epochs: 50
  early_stopping:
    patience: 10
    metric: val_rmse

evaluation:
  metrics:
    - rmse
    - mae
    - mape
    - clarke_error_grid
  test_sets:
    - temporal_holdout
    - patient_holdout
```

**Comment by RobotPsychologist** [comment]
We did not end up following the overall plan on this one, however most of the functionality is there. The planning around this was way to rigid around the interconnections of these modules. Data preparation, training, evaluation, and experimental design do not need to configured interdependently.  One model can be evaluated on multiple experiments for instance. As long as the holdout data is adhered to without leakage we are fine here.
